jakttest/os.cpp

```diff
@@ -22,12 +31,39 @@ ErrorOr<size_t> get_num_cpus()
 ErrorOr<String> get_system_temporary_directory()
 {
     auto builder = TRY(StringBuilder::create());
-#ifdef __WIN32
-    TRY(builder.append_c_string(getenv("TEMP")));
-#else
     auto const result = getenv("TMP_DIR");
-    TRY(builder.append_c_string(result == NULL ? "/tmp" : result));
-#endif
+    TRY(builder.append_c_string(result ?: "/tmp"));
     return TRY(builder.to_string());
 }
+ErrorOr<void> ignore_sigchild()
+{
+    struct sigaction action { };
+    action.sa_handler = [](int) {};
+    sigemptyset(&action.sa_mask);
+    sigaddset(&action.sa_mask, SIGCHLD);
+    action.sa_flags = SA_NOCLDSTOP;
+    if (sigaction(SIGCHLD, &action, NULL) == -1)
+        return Error::from_errno(errno);
+    return {};
+}
+#else
+ErrorOr<size_t> get_num_cpus()
+{
+    SYSTEM_INFO sys_info = {};
+    GetSystemInfo(&sys_info);
+    return __builtin_popcountl(sys_info.dwActiveProcessorMask);
```

Comment:
Can this just be:
```suggestion
    SYSTEM_INFO sys_info = {};
    GetSystemInfo(&sys_info);
    return sys_info.dwNumberOfProcessors;
```
(This seems to be the implementation used in [Boost.Thread](https://github.com/boostorg/thread/blob/6fca28bfed46ee89822d4fffc0f6e7386e98c7b1/src/win32/thread.cpp#L508))

Reply:
The wording of the description on the Microsoft docs is a little confusing, but think the "To retrieve this value, use the [GetLogicalProcessorInformation](https://docs.microsoft.com/en-us/windows/desktop/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformation) function." bit is referring to the processor group.

Reply:
ah, looks like I definitely misread the docs here then, I trust boost on this one.